cmd/image/info/info.go

```diff
@@ -0,0 +1,211 @@
+	archive, err := gzip.NewReader(reader)
+	if err != nil {
+		return err
+	}
+	defer archive.Close()
+
+	writer, err := os.Create(dest)
+	if err != nil {
+		return err
+	}
+	defer writer.Close()
+
+	_, err = io.Copy(writer, archive)
+	return err
+}
+
+// isGzip returns if file is in gzip format
+func isGzip(source string) (bool, error) {
+	file, err := os.Open(source)
+	if err != nil {
+		return false, err
+	}
+	defer file.Close()
+
+	buff := make([]byte, 512)
+	_, err = file.Read(buff)
+	if err != nil {
+		return false, err
+	}
+
+	if filetype := http.DetectContentType(buff); filetype == "application/x-gzip" {
+		return true, nil
+	} else {
+		return false, nil
+	}
+}
+
+func SanitizeExtractPath(filePath string, destination string) error {
+	destpath := filepath.Join(destination, filePath)
+	if !strings.HasPrefix(destpath, filepath.Clean(destination)+string(os.PathSeparator)) {
+		return fmt.Errorf("%s: illegal file path", filePath)
+	}
+	return nil
+}
+
+// Extract specific file from the tar file
+func Untar(tarball, target, filename string) error {
+	reader, err := os.Open(tarball)
+	if err != nil {
+		return err
+	}
+	defer reader.Close()
+	tarReader := tar.NewReader(reader)
+
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		} else if err != nil {
+			return err
+		}
+
+		if header.Name != filename {
+			continue
+		}
+		err = SanitizeExtractPath(header.Name, target)
+		if err != nil {
+			return err
+		}
+		path := filepath.Join(target, header.Name)
+		info := header.FileInfo()
+		if info.IsDir() {
+			if err = os.MkdirAll(path, info.Mode()); err != nil {
+				return err
+			}
+			continue
+		}
+
+		file, err := os.OpenFile(path, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, info.Mode())
+		if err != nil {
+			return err
+		}
+		defer file.Close()
+		_, err = io.Copy(file, tarReader)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+var Cmd = &cobra.Command{
+	Use:   "info",
+	Short: "Provide info about the image",
+	Long: `Provide info about the image
+pvsadm image info --help for information
+
+
+Examples:
+
+# To get the pvsadm tool version by specifying the image
+pvsadm image info --file rhcos-46-12152021.ova.gz
+
+`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		opt := pkg.ImageCMDOptions
+		var ova string
+		ovaImgDir, err := os.MkdirTemp(opt.TemDir, "ova-img-dir")
+		if err != nil {
+			return err
+		}
+
+		defer os.RemoveAll(ovaImgDir)
+
+		//Check if the image is in gzip format and unzip it.
+		checkGzip, err := isGzip(opt.Filename)
+		if err != nil {
+			return fmt.Errorf("failed to detect the image filetype: %v", err)
+		}
+		if checkGzip {
+			klog.Infof("Image %s is in gzip format, extracting it", opt.Filename)
+			ova = filepath.Join(ovaImgDir, "image.ova")
+			err = gunzipIt(opt.Filename, ova)
+			if err != nil {
+				return err
+			}
+			klog.Infof("Extract complete")
+		} else {
+			ova = opt.Filename
+		}
+
+		//Extract the ovf file
+		err = Untar(ova, ovaImgDir, "coreos.ovf")
```

Comment:
I see we have hardcoded these filenames while creating the ova but I prefer making it more generic where tool can feed this based on the type of os image we are creating(out of scope for this PR, lets create a separate issue to track this one).

Wondering if we can untar the file based on the pattern like `*.ovf` will be better to accommodate above change?

Reply:
Done. Modified untar function to untar based on pattern or filename.